portal-wire-protocol.md

```diff
@@ -0,0 +1,161 @@
+# Portal Wire Protocol
+
+The Portal wire protocol is the base p2p protocol by which nodes communicate.
+
+The different Portal networks **MAY** use this base protocol, but they **MUST** remain separated per network.
+This is done at the [Node Discovery Protocol v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md#talkreq-request-0x05) layer, by providing a different protocol bytestring, per network, in the `TALKREQ` message.
+
+The protocol bytestring in the `TALKREQ` message **MUST** have "portal-" as prefix then followed by the network name (e.g. "portal-state").
+
+Each network using the wire protocol **MUST** specify which messages are supported.
+
+Unsupported messages **SHOULD** receive a `TALKRESP` message with an empty payload.
+
+
+## Messages
+
+All messages in the protocol are transmitted using the `TALKREQ` and `TALKRESP` messages from the base protocol.
+
+All messages have a `message_id` and `encoded_message` that are concatenated to form the `payload` for either a `TALKREQ` or `TALKRESP` message.
```

Comment:
Because of the usage of SSZ Union, I now use `selector`. Message type indicates the specific SSZ container specified per message. I hope that is clear.